src/macaron/slsa_analyzer/git_url.py

```diff
@@ -337,11 +337,27 @@ def clone_remote_repo(clone_dir: str, url: str) -> Repo | None:
             os.rmdir(clone_dir)
             logger.debug("The clone dir %s is empty. It has been deleted for cloning the repo.", clone_dir)
         except OSError:
-            logger.debug(
-                "The clone dir %s is not empty. Cloning will not be proceeded.",
-                clone_dir,
-            )
-            return None
+            # If the repository has already been cloned, we attempt to update using `git pull`.
```

Comment:
There is an existing logic to fetch the latest changes from origin remote. You can found it [here](https://github.com/oracle/macaron/blob/b65f0dba4826d32fe79adc1694a7d0f57451654c/src/macaron/slsa_analyzer/git_url.py#L169-L197).
The usual workflow is that after this function is run, the [`checkout_repo_target`](https://github.com/oracle/macaron/blob/b65f0dba4826d32fe79adc1694a7d0f57451654c/src/macaron/slsa_analyzer/git_url.py#L117) function will be run. That `checkout_repo_target` will attempt to fetch latest changes, and make sure our local repository is synced with the origin remote (e.g tags, branches, etc.). 

I wonder for which project did this bug happened?

Reply:
Yeah I see that. The problem was that the `checkout_repo_target` function was being called after the Commit Finder, so in cases where a new tag has been added, the fetch wouldn't happen. I've moved the fetch operation now, replacing the pull I added previously.

Reply:
This will happen for any project that has added new tags when compared to a previously cloned local version.